File: 5_multilevel_inheritance.cpp
#include<bits/stdc++.h>
#define ll long long
using namespace std;
class A{
    public:
        string name = "Bob";
        void displayA() {
            cout << "A display" << endl;
            cout << "Name from A class : " << name << endl;
        }
};
class B: public A {
    public:
        void displayB() {
            cout << "B display" << endl;
        }
};
class C: public B{
    public:
        C() {

        }
        C(string name) {
            this->name = name;
        }
        void displayC() {
            cout << "C display" << endl;
            cout << "Name that was inherited from grandparent : " << name << endl;
        }
};
int main(){
    // C obj;
    C obj("Multilevel inheritance");
    obj.displayA();
    obj.displayB();
    obj.displayC();
    
    return 0;
}

File: 5_multilevel_inheritance.py
class A:
    def __init__(self):
        self.name = "Bob"

    def displayA(self):
        print("A display")
        print("Name from A class:", self.name)

class B(A):
    def displayB(self):
        print("B display")

class C(B):
    def __init__(self, name=""):
        self.name = name

    def displayC(self):
        print("C display")
        print("Name that was inherited from grandparent:", self.name)




# obj = C()
obj = C("Multilevel inheritance")
obj.displayA()
obj.displayB()
obj.displayC()
